### PR TITLE
Upload fixes: firefox warning for folder upload + handle requests closing

### PR DESCRIPTION
**Pull request by frankrousseau**
* When an upload request was closed before the upload finished, it destroys the file metadata.
* Display an error modal when a folder is uploaded via Firefox.
* Add tests for zip route (extract and verify content of downloaded zip)
* Add a button to cancel current upload
* Make spinners more consistent
* Fix progress bar vanishing

### CODE REVIEW

**Comment by frankrousseau** [comment]
I didn't manage to write a test for the aborted uploads. I didn't succeded in writing a fake readable filestream that can be interrupted.

File: package.json [hunk 1]
```diff
@@ -19,26 +19,29 @@
...
-    "jade": "1.3.1",
-    "archiver": "0.10.1",
-    "printit": "0.1.3",
     "cozy-clearance": "0.1.18",
     "cozy-notifications-helper": "1.0.2",
-    "node-polyglot": "0.3.0",
-    "request-json": "0.4.10",
+    "cozy-realtime-adapter": "0.11.3",
+    "cozydb": "0.0.5",
+    "is-absolute": "^0.1.5",
```
Comment (aenario) [inline package.json]: I dont see where is-absolute is used.

File: test/folders_test.coffee [hunk 2]
```diff
@@ -287,19 +291,63 @@ describe "Folders management", ->
...
+            urlPath = "folders/#{@id}/zip/folder"
+            client.saveFile urlPath, archivePath, (err, res, body) =>
                 @res = res
-                done()
+
+                decompress = new Decompress()
+                    .src(archivePath)
+                    .dest(archiveDest)
+                    .use Decompress.zip(strip: 0)
+                setTimeout =>
```
Comment (aenario) [inline test/folders_test.coffee]: please comment on why this timeout.

File: server/controllers/files.coffee [hunk 3]
```diff
@@ -132,124 +132,136 @@ module.exports.create = (req, res, next) ->
...
-                        res.send file, 200
-
-        now = moment().toISOString()
-
-        # Check that the file doesn't exist yet.
-        path = normalizePath path
-        fullPath = "#{path}/#{name}"
-        File.byFullPath key: fullPath, (err, sameFiles) =>
-            return next err if err
+        else
```
Comment (aenario) [inline server/controllers/files.coffee]: This whole change is barely reviewable. We really ought to split this function.

**Comment by frankrousseau** [comment]
I would like to handle folder creation case, before the merge.

File: package.json (hunk 1 above)
Comment (frankrousseau) [inline package.json]: It's an error. I will remove it.

File: test/folders_test.coffee (hunk 2 above)
Comment (frankrousseau) [inline test/folders_test.coffee]: Ok

File: server/controllers/files.coffee [hunk 4]
```diff
@@ -132,124 +132,136 @@ module.exports.create = (req, res, next) ->
...
+                        File.create data, (err, newFile) =>
+                            return next err if err
+
+                            # Ask for the data system to not run autostop
+                            # while the upload is running.
+                            keepAlive()
+
+                            # If user stops the upload, the file is deleted.
+                            err = new Error 'Request canceled by user'
+                            res.on 'close', ->
```
Comment (aenario) [inline server/controllers/files.coffee]: are you sure there is no close event if the request is successful ?

File: server/controllers/files.coffee (hunk 3 above)
Comment (frankrousseau) [inline server/controllers/files.coffee]: Yes, but it's a tough work. Because any wrong assignation can lead to the full loading of the file in memory.

**Comment by aenario** [comment]
Ok, please rebase or merge cozy/master so we can merge from github & build when you are done

File: server/controllers/files.coffee (hunk 4 above)
Comment (frankrousseau) [inline server/controllers/files.coffee]: When the request is succesful It emits an end event.

Comment (frankrousseau) [inline server/controllers/files.coffee]: http://nodejs.org/api/http.html#http_event_close_1

Comment (aenario) [inline server/controllers/files.coffee]: Ok, my bad.

File: server/controllers/files.coffee [hunk 5]
```diff
@@ -244,25 +246,64 @@ module.exports.create = (req, res, next) ->
...
+                            code: 'EEXISTS'
+                            msg: "This file already exists"
+                        , 400
+
+                # Generate file metadata.
+                data =
+                    name: name
+                    path: normalizePath path
+                    creationDate: now
+                    lastModification: now
```
Comment (aenario) [inline server/controllers/files.coffee]: should be lastModification

**Comment by aenario** [comment]
It appears something went wrong while merging cozy/master and it broke the lastModification changes.
(see comment above).

**Comment by frankrousseau** [comment]
Can you handle it?

**Comment by frankrousseau** [comment]
It's cool the right test broke.

**Comment by aenario** [comment]
yes :-)
You need to build

**Comment by frankrousseau** [comment]
Arg the test is still broken

**Comment by frankrousseau** [comment]
Thank  you @aenario for the review.

**Pull request merged**